### ISSUE TITLE
[META] Documentation for extensions

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/security#3354]
This issue serves as a tracker for all documentation tasks for work done that is related to extensions. This tasks need to be completed before extensions are made generally available.

- [ ] https://github.com/opensearch-project/security/issues/2554
- [ ] https://github.com/opensearch-project/security/issues/2709
- [ ] https://github.com/opensearch-project/security/issues/3290

<br/>
_Note; If you see a documentation task that is related to extensions but is not part of this list, please feel free to add it here._

**Comment by scrawfor99** [comment]
[Triage] Thank you for filing this issue @DarshitChanpura. Is this a requirement for the general release, just something to accomplish, something associated with some review? Otherwise looks good.

**Comment by scrawfor99** [comment]
This can be marked triaged....
